lib/index.d.ts

```diff
@@ -72,27 +55,34 @@ export type ClientInstance = {
     }>
 };
 
-export type GlobalConfig = {
-    base?: {
-        // url
-        protocol?: string;
-        // resiliency
-        connectTimeout?: number;
-        maxConnectRetry?: number;
-        timeout?: number;
-        maxFailures?: number; // circuit breaking
-        resetTime?: number; // circuit breaking
-        // agent options
-        agentOptions?: {
-            keepAlive?: boolean;
-            keepAliveMsecs?: number;
-        };
+export type ServiceConfig = {
+    protocol?: string;
+    hostname?: string;
```

Comment:
Removed comments and added types for options that were previously missed like `hostname`